scrap_adidas/spiders/adidas.py

```diff
@@ -0,0 +1,98 @@
+import scrapy
+import logging
+import re
+from scrapy.utils.log import configure_logging
+from scrap_adidas.items import ProductItem, VariationItem, SizeItem
+from copy import deepcopy
+
+class AdidasSpider(scrapy.Spider):
+    
+    name= "adidas"
+    start_urls = [ 
+        'https://shop.adidas.co.id', 
+    ]
+    
+    custom_settings = {
+        'LOG_STDOUT': True,
+    }
+    configure_logging(settings=None, install_root_handler=True)
+    
+    def parse(self, response):
+        url= response.xpath("//li[@class='level0 nav-5 last parent']/a/@href").extract_first()
+        yield scrapy.Request(url, callback= self.parse_product_pages)
+            
+    def parse_product_pages(self, response):
+        product_urls = response.xpath('//h2[@class="product-name"]/a/@href').extract()
+        for url in product_urls:
+            yield scrapy.Request(url, callback= self.product_info)
+            
+        next_page_url = response.xpath('//li[@class="next-page"]/a/@href').extract_first()
+        if next_page_url is not None:
+            yield response.follow(next_page_url, callback= self.parse_product_pages)
+            
+    def product_info(self,response):
```

Comment:
Keep things consistant. `parse_product_info` would be better name instead of `product_info`